pytorch_lightning/trainer/connectors/training_trick_connector.py

```diff
@@ -28,10 +28,21 @@ def on_trainer_init(
         gradient_clip_algorithm: str,
         track_grad_norm: Union[int, float, str],
         accumulate_grad_batches: Union[int, Dict[int, int]],
-        terminate_on_nan: bool,
+        terminate_on_nan: Optional[bool],
```

Comment:
swap order for user using positional arguments and set the deprecated as None by default

Reply:
Eventually you can use `deprecate` wrapper